python/cudf/cudf/utils/dtypes.py

```diff
@@ -634,6 +634,19 @@ def find_common_type(dtypes):
     return cudf.dtype(common_dtype)
 
 
+def common_dtype_compatible(dtypes):
```

Comment:
This isn't being used in `_join_helpers.py` ... is there a way to refactor this to be more common? e.g. Just a passthrough like
```python
def _dtype_pandas_compatible(dtype):
    if cudf.get_option(
        "mode.pandas_compatible"
    ) and dtype == cudf.dtype("O"):
        return "str"
    return dtype
```
Then it doesn't include the `find_common_dtype` logic, and that'd be usable in the `_join_helpers.py` file.

Reply:
Done 👍